### PR TITLE
feat: TE7-T1022: Implement Custom Header

### PR DESCRIPTION
**Pull request by jigar-s-simform**
Implement custom header for some of the screen

### CODE REVIEW

File: app/components/custom-header-home/CustomHeaderHome.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { NavigationRoutes, Strings } from '../../constants';
+import { moderateScale } from '../../theme';
+import { navigateWithParam, toggleDrawer } from '../../utils';
+import styles from './CustomHeaderStyles';
+
+/**
+ * CustomHeader Function
+ *
```
Comment (amitkajwani-simformsolutions) [inline app/components/custom-header-home/CustomHeaderHome.tsx]: remove extra line

File: app/components/custom-header-home/CustomHeaderHome.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * @returns {JSX.Element} - The custom header component for navigation.
+ *
```
Comment (amitkajwani-simformsolutions) [inline app/components/custom-header-home/CustomHeaderHome.tsx:13]: remove extra line

File: app/components/custom-header-home/CustomHeaderHome.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * @description This function is used to define a custom header component for navigation. It takes in an object
+ * with the following parameters: navigation (the navigation object), options (header options), and route (current route).
+ * The function returns a JSX element representing the custom header component.
+ */
+
+
+
```
Comment (amitkajwani-simformsolutions) [inline app/components/custom-header-home/CustomHeaderHome.tsx]: remove extra lines

File: app/components/custom-header-home/CustomHeaderHome.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+const CustomHeaderHome = (): JSX.Element => {
```
Comment (amitkajwani-simformsolutions) [inline app/components/custom-header-home/CustomHeaderHome.tsx:18]: As name suggest 'CustomHeaderHome' so it is used for home screen only?

File: app/constants/Strings.ts [hunk 5]
```diff
@@ -46,4 +46,14 @@ export default {
   opacity:'AA',
   phone: '909900909',
   sendSms: 'sms:+909900909',
+<<<<<<< HEAD
```
Comment (amitkajwani-simformsolutions) [inline app/constants/Strings.ts]: conflicts

File: app/navigation/DashboardDrawer.tsx [hunk 6]
```diff
@@ -1,6 +1,10 @@
 import { createDrawerNavigator } from '@react-navigation/drawer';
 import { FC } from 'react';
+<<<<<<< HEAD
```
Comment (amitkajwani-simformsolutions) [inline app/navigation/DashboardDrawer.tsx]: conflicts

File: app/navigation/home-tab/HomeTabButton.tsx [hunk 7]
```diff
@@ -16,16 +16,16 @@ const TabButton = (props: TabButtonProps): JSX.Element => {
...
-                useNativeDriver: true,
-            }).start();
-        } else {
-            yOffest.setValue(0);
-        }
-    }, [focused]);
+  useEffect(() => {
+    if (focused) {
+      Animated.spring(yOffest, {
+        toValue: -moderateScale(10),
```
Comment (amitkajwani-simformsolutions) [inline app/navigation/home-tab/HomeTabButton.tsx:22]: why minus value?

File: app/navigation/home-tab/HomeTabs.tsx [hunk 8]
```diff
@@ -14,11 +14,11 @@ const HomeTabs: FC = () => {
   return (
     <Tabs.Navigator
       screenOptions={{
-        headerShown: false,
         tabBarStyle: styles.navigatorStyles,
       }}>
       <Tabs.Screen
         options={{
+          headerShown:false,
```
Comment (amitkajwani-simformsolutions) [inline app/navigation/home-tab/HomeTabs.tsx]: add space after :

File: app/navigation/HomeStack.tsx [hunk 9]
```diff
@@ -9,8 +10,11 @@ const HomeStackNavigator = createNativeStackNavigator<HomeStackParamsList>();
 const HomeStack:FC = () => {
```
Comment (amitkajwani-simformsolutions) [inline app/navigation/HomeStack.tsx]: space after :

File: app/navigation/DashboardDrawer.tsx [hunk 10]
```diff
@@ -23,6 +27,9 @@ const DashBoardDrawer: FC = (): JSX.Element => {
   return (
     <DasboardDrawer.Navigator>
       <DasboardDrawer.Screen
+        options={{
+          headerShown:false
```
Comment (amitkajwani-simformsolutions) [inline app/navigation/DashboardDrawer.tsx]: space after :

**Review by amitkajwani-simformsolutions — CHANGES_REQUESTED** [review]

File: app/navigation/home-tab/HomeTabButton.tsx (hunk 7 above)
Comment (jigar-s-simform) [reply app/navigation/home-tab/HomeTabButton.tsx:22]: it is required to create a bouncing effect for the active screen icon

File: app/components/custom-header-home/CustomHeaderHome.tsx (hunk 4 above)
Comment (jigar-s-simform) [reply app/components/custom-header-home/CustomHeaderHome.tsx:18]: yes

Comment (amitkajwani-simformsolutions) [reply app/components/custom-header-home/CustomHeaderHome.tsx:18]: Can't we use the same for all tabs?

**Review by amitkajwani-simformsolutions — CHANGES_REQUESTED** [review]

File: app/components/custom-header-home/CustomHeaderHome.tsx (hunk 4 above)
Comment (jigar-s-simform) [reply app/components/custom-header-home/CustomHeaderHome.tsx:18]: No it is specifically for home tab!

Comment (amitkajwani-simformsolutions) [reply app/components/custom-header-home/CustomHeaderHome.tsx:18]: Okay!

**Review by amitkajwani-simformsolutions — APPROVED** [review]

**Pull request merged**